vernac/g_vernac.mlg

```diff
@@ -404,13 +407,16 @@ GRAMMAR EXTEND Gram
   ;
   constructors_or_record:
     [ [ "|"; l = LIST1 constructor SEP "|" -> { Constructors l }
-      | id = identref ; c = constructor_type; "|"; l = LIST1 constructor SEP "|" ->
-          { Constructors ((c id)::l) }
-      | id = identref ; c = constructor_type -> { Constructors [ c id ] }
-      | cstr = identref; "{"; fs = record_fields; "}"; id = default_inhabitant_ident ->
-          { RecordDecl (Some cstr,fs,id) }
-      | "{";fs = record_fields; "}"; id = default_inhabitant_ident -> { RecordDecl (None,fs,id) }
-      |  -> { Constructors [] } ] ]
+      | attr = LIST0 quoted_attributes ; id = identref ; c = constructor_type;
+        "|"; l = LIST1 constructor SEP "|" ->
+          { Constructors ((c attr id)::l) }
+      | attr = LIST0 quoted_attributes ; id = identref ; c = constructor_type ->
+          { Constructors [ c attr id ] }
+      | attr = LIST0 quoted_attributes ;
+        cstr = identref; "{"; fs = record_fields; "}"; id = default_inhabitant_ident ->
+          { let () = parse_attr (Notations.return ()) attr in
```

Comment:
What's going on here? Did you mean 
```suggestion
          { let () = unsupported_attributes attr in
```
?

Reply:
Indeed, that's much better, thanks!